### PR TITLE
Add a Node platform.

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

**Comment by natebosch** [comment]
Could we add more details to the commit message?

**Comment by nex3** [comment]
Sure! What are you looking for in particular?

**Comment by natebosch** [comment]
Mainly hoping for a breakdown of what are changes to existing extension points to support this vs the pieces that would go in for any new platform.

**Comment by nex3** [comment]
Done.

File: lib/src/runner/node/platform.dart [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      var response = await (await _http.getUrl(url)).close();
+
+      if (response.statusCode != 200) {
+        // We don't care about the response body, but we have to drain it or
+        // else the process can't exit.
+        response.listen(null);
+
+        throw new LoadException(
+            suitePath,
+            "Error getting $url: ${response.statusCode} "
```
Comment (efortuna) [inline lib/src/runner/node/platform.dart]: consider using single quotes here and below

File: lib/src/runner/node/platform.dart [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            "${response.reasonPhrase}\n"
+            'Make sure "pub serve" is serving the test/ directory.');
+      }
+
+      return await UTF8.decodeStream(response);
+    } on IOException catch (error) {
+      var message = getErrorMessage(error);
+      if (error is SocketException) {
+        message = "${error.osError.message} "
```
Comment (efortuna) [inline lib/src/runner/node/platform.dart]: okay I'm going to stop posting about this quote thing, but check out other places in these files that it's happening.

**Comment by nex3** [comment]
I think the styleguide intentionally *doesn't* provide guidance on string quotes, because keeping them consistent is a lot of overhead for little benefit. This package certainly doesn't strive for consistency in general. I'd prefer to leave them as-is.

**Pull request merged**